src/components/grid/grid.pw.tsx

```diff
@@ -0,0 +1,366 @@
+
+      await expect(expectedGridItem).toHaveCSS("grid-row", `${row}`);
+      await expect(expectedGridItem).toHaveCSS("grid-column", `${col}`);
+      await assertCssValueIsApproximately(
+        expectedGridItem,
+        "width",
+        Number(expectedWidth)
+      );
+    });
+  });
+
+  [
+    [1, "auto / auto", "1 / 13", 1194],
+    [2, "2 / 3", "1 / 6", 483],
+    [3, "4 / 5", "7 / 13", 585],
+  ].forEach(([itemNumber, row, col, expectedWidth]) => {
+    test(`when viewport size is medium, grid item ${itemNumber} should have correct grid-row, grid-column and width`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<GridLayoutExample />);
+      await page.setViewportSize({ width: 1259, height: 900 });
+      const expectedGridItem = gridItem(page, Number(itemNumber) - 1);
+
+      await expect(expectedGridItem).toHaveCSS("grid-row", `${row}`);
+      await expect(expectedGridItem).toHaveCSS("grid-column", `${col}`);
+      await assertCssValueIsApproximately(
+        expectedGridItem,
+        "width",
+        Number(expectedWidth)
+      );
+    });
+  });
+
+  [
+    [1, "auto / auto", "1 / 13", 1842],
+    [2, "2 / 3", "1 / 6", 752],
+    [3, "4 / 5", "7 / 13", 906],
+  ].forEach(([itemNumber, row, col, expectedWidth]) => {
+    test(`when viewport size is large, grid item ${itemNumber} should have correct grid-row, grid-column and width`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<GridLayoutExample />);
+      await page.setViewportSize({ width: 1920, height: 900 });
+      const expectedGridItem = gridItem(page, Number(itemNumber) - 1);
+
+      await expect(expectedGridItem).toHaveCSS("grid-row", `${row}`);
+      await expect(expectedGridItem).toHaveCSS("grid-column", `${col}`);
+      await assertCssValueIsApproximately(
+        expectedGridItem,
+        "width",
+        Number(expectedWidth)
+      );
+    });
+  });
+
+  [
+    [1, "auto / auto", "1 / 13", 1842],
+    [2, "2 / 3", "1 / 6", 744],
+    [3, "4 / 5", "7 / 13", 901],
+  ].forEach(([itemNumber, row, col, expectedWidth]) => {
+    test(`when viewport size is extra large, grid item ${itemNumber} should have correct grid-row, grid-column and width`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<GridLayoutExample />);
+      await page.setViewportSize({ width: 1922, height: 900 });
+      const expectedGridItem = gridItem(page, Number(itemNumber) - 1);
+
+      await expect(expectedGridItem).toHaveCSS("grid-row", `${row}`);
+      await expect(expectedGridItem).toHaveCSS("grid-column", `${col}`);
+      await assertCssValueIsApproximately(
+        expectedGridItem,
+        "width",
+        Number(expectedWidth)
+      );
+    });
+  });
+
+  [
+    ["extra small", 16, 16],
+    ["small", 24, 16],
+    ["medium", 32, 24],
+    ["large", 40, 24],
+    ["extra large", 40, 40],
+  ].forEach(([screenSize, padding, gridGap]) => {
```

Comment:
**suggestion**: what do you think about including the viewport dimensions as test parameters instead of using the switch statement on line 167?

```suggestion
    ["extra small", 599, 900, 16, 16],
    ["small", 959, 900, 24, 16],
    ["medium", 1259, 900, 32, 24],
    ["large", 1920, 900, 40, 24],
    ["extra large", 1922, 900, 40, 40],
  ].forEach(([screenSize, viewportHeight, viewportWidth, padding, gridGap]) => {
```